### ISSUE TITLE
Minecraft freezes at "saving world" in offline mode if player name is changed.

### ISSUE THREAD

**Issue by chris000000** [MultiMC/MultiMC5#3620]
System Information
-----------------------------
MultiMC version: 
0.6.11-1430

Operating System:
Windows 10

Summary of the issue or suggestion:
----------------------------------------------
If a world is launched in offline mode (since around Minecraft 1.14)  and the player name is changed in the player name window, this results in Minecraft freezing at the "saving world" screen when I have finished playing and try to save the game. If I uninstall Forge then the game saves just fine. 

In addition, in-game my player name in chat is my account name not the changed name, if I uninstall Forge then my player name in chat is the changed name. 

What should happen:
------------------------------
The game should just save and go back to the Minecraft main menu.  The player name in chat should be the changed player name.

Steps to reproduce the issue (Add more if needed):
-------------------------------------------------------------
1. Launch a 1.16.5 offline instance with latest Forge installed

2. Change the name in the player name box that pops up

3. When in a world press T and type a message, it will display your Minecraft account name not the changed name.

4. At the end of play "save and quit to title" and it will stick at the "saving world screen.

Suspected cause:
---------------------------
It happens on 1.16.5 instance but doesn't happen on 1.12.2 or 1.7.10 instance ; maybe it is to do with a change in Minecraft or Forge.  

Logs/Screenshots:
----------------------------

Additional Info:
---------------------------

**Comment by Forkk** [comment]
Please reproduce the issue and provide the logs. We cannot diagnose anything without them.

**Comment by peterix** [comment]
Then don't change the name...

**Comment by chris000000** [comment]
Logs here, it can be reproduced by following the steps above.
[debug.log](https://github.com/MultiMC/MultiMC5/files/6096292/debug.log)
[MultiMC-0.log](https://github.com/MultiMC/MultiMC5/files/6096293/MultiMC-0.log)

Changing the name is a feature we use when launching offline, and I'm just pointing out an issue with it. Currently I don't change the name because the feature doesn't work properly. If it is a forge issue that can't be fixed maybe you will want to remove the option that allows name change, but I am hoping it can be fixed.

**Comment by peterix** [comment]
No, I do not think I can fix a game bug. And I will not remove this just because it breaks in some particular version of the game or Forge.

The game launches, and that is where MultiMC's involvement ends (aside from watching the game's process for logs/crashes).

**Comment by chris000000** [comment]
Yes, I see your point, it does work without forge installed and for mc versions older than ~1.14 with forge installed so the feature could stll be useful to some people.  I don't think this is going correct itself in the future though because the forge team won't or maybe can't fix it (https://forums.minecraftforge.net/topic/98059-minecraft-freezes-at-saving-world/?tab=comments#comment-444483). Thanks for responding and for making MultiMC, it is pretty awesome.

**Comment by peterix** [comment]
It's a vanilla bug if it happens with vanilla, a Forge bug if it happens with Forge alone, or a mod bug if it happens with Forge and mods installed.

The 'being launched through a hack' part in that exchange is pure nonsense. It changes nothing about the game saving or not saving worlds. The "hack" is absolutely necessary because Forge doesn't provide any good way to install it. The forge people are simply trying to dismiss the issue based on irrelevant unrelated details.

**Comment by peterix** [comment]
So the real question is: where is the root cause?

I assume not MultiMC. It's probably someone else not accounting for uncommonly used launch parameters.

**Comment by chris000000** [comment]
It is probably just something that has fallen between the cracks of how vanilla/forge/multimc interact, and if I were to guess I would guess the root cause is probably a change to vanilla around 1.14 which forge only mitigated for certain common scenarios. Anyway, like you say, it will be easiest if I just don't change the player name. Thanks again.

**Comment by eikesauer** [comment]
For the records, I have the same problem, under Linux, using Fabric.

**Comment by paw-all-obo-alexaro-std** [comment]
I also have this problem, but with vanilla minecraft.
